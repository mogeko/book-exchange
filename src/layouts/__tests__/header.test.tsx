import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Header from "@/layouts/header";

describe("Header", () => {
  it("should render correctly", () => {
    const { container } = render(<Header />);

    expect(
      screen.getByRole("heading", { name: /Bookworm/i })
    ).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
