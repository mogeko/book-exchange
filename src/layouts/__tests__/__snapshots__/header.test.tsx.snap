// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div>
  <div
    class="min-w-full"
  >
    <div
      class="hidden h-7 bg-black justify-between sm:flex"
      data-theme="dracula"
    >
      <nav
        class="navbar min-h-0 flex-"
      >
        <a
          class="btn btn-link btn-xs gap-1 text-gray-400 hover:text-primary focus:text-primary-focus"
          href="/#"
        >
          My library
        </a>
        <a
          class="btn btn-link btn-xs gap-1 text-gray-400 hover:text-primary focus:text-primary-focus"
          href="/#"
        >
          Ranking
        </a>
        <a
          class="btn btn-link btn-xs gap-1 text-gray-400 hover:text-primary focus:text-primary-focus"
          href="/#"
        >
          Quote
        </a>
        <a
          class="btn btn-link btn-xs text-gray-400 hover:text-primary focus:text-primary-focus"
          href="/random"
        >
          I'm Feeling Lucky!
        </a>
      </nav>
      <div
        class="justify-end"
      >
        <div
          class="dropdown dropdown-end"
        >
          <label
            class="hidden sm:block normal-case"
            tabindex="0"
          >
            <div
              class="animate-pulse w-20 mr-4 h-2 bg-slate-700 rounded"
            />
          </label>
          <label
            class="btn btn-ghost btn-circle avatar sm:hidden"
            tabindex="0"
          >
            <div
              class="w-6 rounded-full"
            >
              <svg
                class="animate-spin w-6 h-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M13.917 7A6.002 6.002 0 0 0 2.083 7H1.071a7.002 7.002 0 0 1 13.858 0h-1.012z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </label>
          <ul
            aria-label="User Menu"
            class="menu menu-compact dropdown-content shadow bg-base-300 rounded-box w-52 mt-3 p-2 sm:mt-4 sm:p-0"
            role="menu"
            tabindex="0"
          >
            <li
              aria-label="User Menu Item"
              role="menuitem"
            >
              <a
                class="justify-between"
                href="/#"
              >
                <span>
                  Profile
                </span>
              </a>
            </li>
            <li
              aria-label="User Menu Item"
              role="menuitem"
            >
              <a
                class="justify-between"
                href="/#"
              >
                <span>
                  Settings
                </span>
              </a>
            </li>
            <li
              aria-label="User Menu Item"
              role="menuitem"
            >
              <button>
                Sign out
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="bg-inherit p-2 md:p-0 md:bg-base-300"
    >
      <header
        class="navbar p-0 m-auto rounded-md max-w-screen-lg bg-base-300 md:rounded-none"
      >
        <div
          class="navbar-start sm:hidden"
        >
          <label
            aria-label="Drawer Button"
            class="btn btn-ghost btn-circle"
            for="mobile-menu"
            role="button"
          >
            <svg
              aria-hidden="true"
              class="h-6 w-6"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                fill-rule="evenodd"
              />
            </svg>
          </label>
        </div>
        <div
          class="navbar-center sm:navbar-start"
        >
          <div
            class="flex justify-start"
          >
            <a
              class="btn gap-1 btn-ghost normal-case"
              href="/"
            >
              <span
                class="sr-only"
              >
                Logo
              </span>
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="logo"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
              <h1
                class="text-xl font-bold text-base-content"
              >
                Bookworm
              </h1>
            </a>
          </div>
        </div>
        <div
          class="navbar-end"
        >
          <div
            class="hidden md:block relative"
          >
            <svg
              class="h-6 w-6 absolute top-3 left-3 text-gray-400"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
                fill-rule="evenodd"
              />
            </svg>
            <div
              class="form-control"
            >
              <input
                aria-label="Search Bar"
                class="input bg-base-300 text-xl w-96 pl-10"
                placeholder="Search..."
                role="search"
                type="text"
              />
            </div>
          </div>
          <a
            class="btn btn-ghost btn-circle md:hidden"
            href="/#"
            role="button"
          >
            <svg
              aria-hidden="true"
              class="h-6 w-6"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              class="sr-only"
            >
              Search
            </span>
          </a>
          <div
            class="sm:hidden"
          >
            <div
              class="dropdown dropdown-end"
            >
              <label
                class="hidden sm:block normal-case"
                tabindex="0"
              >
                <div
                  class="animate-pulse w-20 mr-4 h-2 bg-slate-700 rounded"
                />
              </label>
              <label
                class="btn btn-ghost btn-circle avatar sm:hidden"
                tabindex="0"
              >
                <div
                  class="w-6 rounded-full"
                >
                  <svg
                    class="animate-spin w-6 h-6"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M13.917 7A6.002 6.002 0 0 0 2.083 7H1.071a7.002 7.002 0 0 1 13.858 0h-1.012z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </div>
              </label>
              <ul
                aria-label="User Menu"
                class="menu menu-compact dropdown-content shadow bg-base-300 rounded-box w-52 mt-3 p-2 sm:mt-4 sm:p-0"
                role="menu"
                tabindex="0"
              >
                <li
                  aria-label="User Menu Item"
                  role="menuitem"
                >
                  <a
                    class="justify-between"
                    href="/#"
                  >
                    <span>
                      Profile
                    </span>
                  </a>
                </li>
                <li
                  aria-label="User Menu Item"
                  role="menuitem"
                >
                  <a
                    class="justify-between"
                    href="/#"
                  >
                    <span>
                      Settings
                    </span>
                  </a>
                </li>
                <li
                  aria-label="User Menu Item"
                  role="menuitem"
                >
                  <button>
                    Sign out
                  </button>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </header>
    </div>
  </div>
</div>
`;
